lasy/profiles/longitudinal/longitudinal_profile_from_data.py

```diff
@@ -0,0 +1,151 @@
+            intensity. This will be noramlized
+            When datatype is 'temporal' intensity is temporal
+            intensity. This will be normalized
+
+        phase : ndarrays of floats
+            If provided, this phase will be added to the pulse.
+            When datatype is 'spectral' phase is spectral
+            phase.
+            When datatype is 'temporal' phase is temporal
+            phase
+
+        dt : float
+            Only required when datatype is 'spectral'. In this
+            case this defines the user requested resolution in
+            the conversrion from the spectral to the temporal
+            domain.
+
+        wavelength : float
+            Only required when datatype is 'temporal'. Then,
+            this is the central wavelength of the pulse
+
+    lo, hi : floats (seconds)
+        Lower and higher end of the required domain of the data.
+        The data imported will be cut to this range prior to
+        being incorporated into the ``lasy`` pulse.
+    """
+
+    def __init__(self, data, lo, hi):
+        if data["datatype"] == "spectral":
+            # First find central frequency
+            wavelength = data["abscissa"]
+            spectral_intensity = data["intensity"]
+            spectral_phase = data["phase"]
+            dt = data["dt"]
+            cwl = np.sum(spectral_intensity * wavelength) / np.sum(spectral_intensity)
+            cfreq = c / cwl
+            # Determine required sampling frequency for desired dt
+            sample_freq = 1 / dt
+            # Determine number of points in temporal domain
+            indx = np.argmin(np.abs(wavelength - cwl))
+            dfreq = np.abs(c / wavelength[indx] - c / wavelength[indx + 1])
+            N = int((sample_freq) / dfreq)
+            freq = np.linspace(cfreq - sample_freq / 2, cfreq + sample_freq / 2, N)
+            # interpolate the spectrum onto this new array
+            freq_intensity = np.interp(
+                freq, c / wavelength[::-1], spectral_intensity[::-1], left=0, right=0
+            )
+            freq_phase = np.interp(
+                freq, c / wavelength[::-1], spectral_phase[::-1], left=0, right=0
+            )
+
+            freq_amplitude = np.sqrt(freq_intensity)
+
+            # Inverse Fourier Transform to the time domain
+            t_amplitude = (
+                np.fft.fftshift(
+                    np.fft.ifft(
+                        np.fft.ifftshift(freq_amplitude * np.exp(-1j * freq_phase))
+                    )
+                )
+                / dt
+            )
+            time = np.linspace(-dt * N / 2, dt * N / 2 - dt, N)
+
+            # Extract intensity and phase
+            temporal_intensity = np.abs(t_amplitude) ** 2
+            temporal_intensity /= np.max(temporal_intensity)
+            temporal_phase = np.unwrap(-np.angle(t_amplitude))
+            temporal_phase -= temporal_phase[np.argmin(np.abs(time))]
```

Comment:
Should the phase be defined with a `-` (as currently done in these 2 lines) or with a `+` sign? I am not sure what the most natural/usual convention is here, considering that we use `exp(-j*omega0*t)` in the definition of the envelope. Note that if we change it here, the `evaluate`function below also needs to be updated.

Reply:
So it has been tested and the intensity lineout extracted from evaluating the longitudinal profile or from indexing directly into `laser.field.field` correctly reproduces the temporal profile (the correct way around, so think the +/- is correct). However, currently we don't match with reading back in the openPMD output, neither in `xyt` or `rt` but I think this is a future problem

<img width="487" alt="Screenshot 2023-05-18 at 17 22 09" src="https://github.com/LASY-org/lasy/assets/27694869/35ff0fe5-8fef-4ae0-9f93-b06739802fd7">

Reply:
Thanks for clarifying!